bindings/java/src/org/sleuthkit/datamodel/HostManager.java

```diff
@@ -208,70 +213,73 @@ public Host createHost(String name) throws TskCoreException  {
 			}
 		} catch (SQLException ex) {
 			// may have failed because it already exists. So try getting the host.
-			 Optional<Host> host = this.getHost(name, connection);
-			 if (host.isPresent()) {
-				 return host.get();
-			 } else {
-				 throw new TskCoreException(String.format("Error adding host with name = %s", name), ex);
-			 }
+			Optional<Host> host = this.getHost(name, connection);
+			if (host.isPresent()) {
+				return host.get();
+			} else {
+				throw new TskCoreException(String.format("Error adding host with name = %s", name), ex);
+			}
 		} finally {
 			db.releaseSingleUserCaseWriteLock();
 		}
 	}
-	
+
 	/**
-	 * Get host with given name. 
-	 * 
-	 * @param name Host name to look for. 
+	 * Get host with given name.
+	 *
+	 * @param name        Host name to look for.
 	 * @param transaction Database transaction to use.
-	 * 
-	 * @return Optional with host.  Optional.empty if no matching host is found.
-	 * 
-	 * @throws TskCoreException 
+	 *
+	 * @return Optional with host. Optional.empty if no matching host is found.
+	 *
+	 * @throws TskCoreException
 	 */
-	public Optional<Host> getHost(String name, CaseDbTransaction transaction ) throws TskCoreException {
-		
+	public Optional<Host> getHost(String name, CaseDbTransaction transaction) throws TskCoreException {
+
 		return getHost(name, transaction.getConnection());
 	}
-	
+
 	/**
-	 * Get host with given name. 
-	 * 
-	 * @param name Host name to look for. 
+	 * Get host with given name.
+	 *
+	 * @param name       Host name to look for.
 	 * @param connection Database connection to use.
-	 * 
-	 * @return Optional with host.  Optional.empty if no matching host is found.
-	 * 
-	 * @throws TskCoreException 
+	 *
+	 * @return Optional with host. Optional.empty if no matching host is found.
+	 *
+	 * @throws TskCoreException
 	 */
-	private Optional<Host> getHost(String name, CaseDbConnection connection ) throws TskCoreException {
-		
+	private Optional<Host> getHost(String name, CaseDbConnection connection) throws TskCoreException {
+
 		String queryString = "SELECT * FROM tsk_hosts"
-							+ " WHERE LOWER(name) = LOWER('" + name + "')";
-		try (
-				Statement s = connection.createStatement();
-				ResultSet rs = connection.executeQuery(s, queryString)) {
+				+ " WHERE LOWER(name) = LOWER(?)";
+		try (PreparedStatement s = connection.getPreparedStatement(queryString, Statement.RETURN_GENERATED_KEYS)) {
```

Comment:
It looks like this try-with-resources is resulting in the transaction being closed when it shouldn't be.